### ISSUE TITLE
model_admin'. as_view only accepts arguments that are already attributes of the class

### ISSUE THREAD

**Issue by adontz** [farhan0581/django-admin-autocomplete-filter#60]
Django==3.2.2
django-admin-autocomplete-filter==0.6.1

After upgrading from Django==3.1.5 I receive

`CustomFilterView() received an invalid keyword 'model_admin'. as_view only accepts arguments that are already attributes of the class.`

For the following code

```
class MyModelAdmin(admin.ModelAdmin):
...
    def get_urls(self):
        base_urls = super().get_urls()
        filter_urls = [
            path(
                'filter-autocomplete/',
                self.admin_site.admin_view(CustomFilterView.as_view(model_admin=self)),
                name='authnz_custom_filter_autocomplete'),
        ]

        return filter_urls + base_urls
```

**Comment by adontz** [comment]
@farhan0581 is this project supported yet?

**Comment by farhan0581** [comment]
can you try using the new version 0.7 ?

**Comment by th-thomas** [comment]
Same issue but with Django 3.2.3 and version 0.7

**Comment by bboru21** [comment]
@adontz and @th-thomas  - 

Looks like Django introduced validation on View attributes within the as_view method:
https://github.com/django/django/blob/main/django/views/generic/base.py#L56-L59

I _think_ adding model_admin as an attribute to the CustomFilterView should get it to pass that validation and get it working for you:

`
class CustomFilterView(AutocompleteJsonView):

    model_admin = None

    def get_queryset(self):
        # ...
`

**Comment by tatterdemalion** [comment]
Adding `model_admin = None` fails because of this line: https://github.com/farhan0581/django-admin-autocomplete-filter/blob/pre_release/admin_auto_filters/views.py#L17

```
self.paginator_class = self.model_admin.paginator
```

I overcame this by setting base model admin class `admin.ModelAdmin` and changing `get_paginator` method:

```
class CustomFilterView(AutocompleteJsonView):
    model_admin = admin.ModelAdmin

    def get_paginator(self, *args, **kwargs):
        return self.paginator_class(*args, **kwargs)
```

I know this is a stupid workaround but seems to get the job done.

**Comment by creshal** [comment]
I tried this workaround, and while it makes Django run without errors, it always returns empty result sets, even if I don't override `get_queryset` at all. Something still seems broken.

**Comment by edu2004eu** [comment]
For those still struggling with this, I've found a workaround. You will need to override the `get` method on the custom search view:

    def get(self, request, *args, **kwargs):
        self.term, self.model_admin, self.source_field, to_field_name = self.process_request(request)

        return super().get(request, *args, **kwargs)

**Comment by shaig-planetly** [comment]
@edu2004eu this causes a 403 error

**Comment by ashishnitinpatil** [comment]
I had to do something hacky in order to resolve all issues & get the view to work correctly -

```
class CustomFilterView(AutocompleteJsonView):
    model_admin = MyAdmin #  using my final class; can get away with None as well

    def get_queryset(self):
        # overriding to fix Django 3.2 upgrade issue (removed self.complex_query line)
        qs = self.model_admin.get_queryset(self.request)
        qs, search_use_distinct = self.model_admin.get_search_results(
            self.request, qs, self.term
        )
        # qs = qs.filter() -> my custom search
        if search_use_distinct:
            qs = qs.distinct()
        return qs
```

I'm using the code from the PR https://github.com/farhan0581/django-admin-autocomplete-filter/pull/78

**Comment by cash** [comment]
The commit in Django 3.2 that broke this package is here: https://github.com/django/django/commit/3071660acfbdf4b5c59457c8e9dc345d5e8894c5

From Django developers' perspective, `AutocompleteJsonView` is private because it is not documented so they can change it without deprecation.

I think many uses cases can be handled by this package: https://github.com/demiroren-teknoloji/django-admin-autocomplete-list-filter

**Comment by mscheper** [comment]
@cash:
> I think many uses cases can be handled by this package

Unfortunately, the version pip installed still uses `ugettext_lazy`, which seems to be long deprecated and not in Django 5. I do see a commit in that repo to work around it, but even that's from 3 years ago, so I don't think that one is being maintained any more, either.

@ashishnitinpatil:
> I had to do something hacky in order to resolve all issues & get the view to work correctly -

When I tried it, I'm seeing no query results, and by adding debugging lines, it seems `get_queryset` isn't even getting called. I suppose this is because that PR hasn't been merged into the code I'm using?

**Comment by ashishnitinpatil** [comment]
Yes, most likely because you aren't using the code from the PR I'm referencing @mscheper 
I think I just added the pip dependency with the specific commit of that PR branch (and made the changes on top)
